extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-tag.js

```diff
@@ -69,6 +84,20 @@ describes.realWin(
       expect(() => shoppingTag.layoutCallback()).to.not.throw();
     });
 
+    it('should not build shopping tag if page attachment is removed', async () => {
+      const createAndAppendInnerShoppingTagElStub = env.sandbox.stub(
+        shoppingTag,
+        'createAndAppendInnerShoppingTagEl_'
+      );
+
+      pageEl.removeChild(attachmentElement);
+
+      await shoppingTag.buildCallback();
+      await shoppingTag.layoutCallback();
+
+      expect(createAndAppendInnerShoppingTagElStub).to.not.be.called;
```

Comment:
As much as possible, you should not stub any private property/method. Tying tests to the implementation makes the class harder to refactor.
Instead, you could check that `shoppingTag.element` is empty, or has a specific class, or anything that works with the implementation details you have here.

cf "Testing via public APIs" in the link I sent you privately

Reply:
I'll take a look at the document more carefully.